src/EditorFeatures/CSharpTest2/Recommendations/ReadOnlyKeywordRecommenderTests.cs

```diff
@@ -588,5 +588,27 @@ void M()
     }
 }");
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        public async Task TestInFunctionPointerTypeAfterRef()
+        {
+            await VerifyKeywordAsync(@"
+class C
+{
+    delegate*<ref $$");
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.KeywordRecommending)]
+        [InlineData("in")]
+        [InlineData("out")]
+        [InlineData("")]
+        [InlineData("ref readonly")]
+        public async Task TestNotInFunctionPointerTypeNoRef(string modifier)
+        {
+            await VerifyAbsenceAsync($@"
+class C
+{{
+    delegate*<{modifier} $$");
```

Comment:
oh, you can't say `delegate*<readonly int x>`?  you have to say `ref readonly` only?

Reply:
i'd honestly prefer a dedicated test for taht.

Reply:
Not sure I understand?

Reply:
Added test from offline conversation.